src/System.Linq.Queryable/src/System/Linq/Queryable.cs

```diff
@@ -39,10 +39,10 @@ public static IQueryable<TSource> Where<TSource>(this IQueryable<TSource> source
             return source.Provider.CreateQuery<TSource>(
                 Expression.Call(
                     null,
-                    GetMethodInfoOf(() => Queryable.Where(
+                    GetMethodInfoOf(() => Where(
```

Comment:
Unrelated to this PR, but we may want to look into the performance implications of `GetMethodInfoOf` without any caching. The construction of an expression tree in order to obtain a (closed generic) `MethodInfo` instance for each and every call to any of the `Queryable` factory methods seems very expensive, both in terms of argument validation in the factories, and allocations of temporary objects. Maybe an approach a la `CachedReflectionInfo` in `System.Linq.Expressions` would make sense here, caching open generic methods for the query operators. I can have a look at that.